backend/api.py

```
@@ -1090,38 +1090,104 @@ def search_molecules():
+        if include_chembl and len(query) >= 2 and len(local_results) < limit:
+            from chembl_drug_search import (
+                search_chembl_by_name,
+                upsert_chembl_hits_into_drugs,
+                merge_drug_search_results,
+            )
+            chembl_hits = search_chembl_by_name(query, limit=limit)
+            chembl_used = True
+            if chembl_hits:
+                # Write-through cache so the next search is fast & local.
+                new_ids = upsert_chembl_hits_into_drugs(chembl_hits)
+                for hit, new_id in zip(chembl_hits, new_ids):
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)